builder/image/templates/builder

```diff
@@ -112,8 +106,10 @@ if [ ! -f Dockerfile ]; then
     BUILD_OPTS+=' -v '
     BUILD_OPTS+=$(echo $CACHE_DIR)
     BUILD_OPTS+=':/tmp/cache:rw'
-    # give non-root slugbuilder user R/W perms for docker volumes
-    chown -R 2000:2000 $TMP_DIR $CACHE_DIR
+    # give slug group ownership of TMP_DIR and CACHE_DIR. This is used by the
+    # slug builder. Contents in these folders will be owned by slug:slug.
```

Comment:
Yes, that seems to be the correct behaviour. `TMP_DIR` is synonymous with a buildpack's `$BUILD_DIR` (IOW, the application's source code), which is copied in. `CACHE_DIR` is used by the buildpack to cache assets between builds, so anything built inside there would be owned by `slug:slug`.

> TMP_DIR is owned by git:slug, but the contents inside the directory is owned by git:git.

That won't be good for some buildpacks. IIRC some require write access to some application's source subdirectories for vendoring in assets like gems.  You'll probably want to make the group permissions recursive.